src/main/java/fi/aalto/cs/apluscourses/intellij/actions/DownloadSubmissionAction.java

```diff
@@ -204,25 +221,33 @@ private void downloadFiles(@Nullable Project project,
                              @NotNull Module module) {
     var courseViewModel = mainViewModelProvider.getMainViewModel(project).courseViewModel.get();
     if (courseViewModel == null || project == null) {
+      logger.info("CourseViewModel or project null");
       return;
     }
     var course = courseViewModel.getModel();
     course.validate();
 
     var auth = authenticationProvider.getAuthentication(project);
 
+    logger.info("Starting download");
+    var stopwatch = new Stopwatch();
     for (var info : submissionFilesInfo) {
       try {
         var file = fileFinder.findFile(module.getFullPath(), info.getFileName()).toFile();
+        logger.info("Found file: {}", file);
         CoursesClient.fetch(new URL(info.getUrl()), file, auth);
+        logger.info("Fetched file");
         fileBrowser.navigateTo(file, project);
+        logger.info("Opened file");
```

Comment:
Only the first log entry is useful. The remaining entries don't give any additional information, and if something goes wrong then an exception is thrown and the log entries in `catch` clauses take over.